webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -742,120 +742,128 @@ class ClassicTechniqueWriter(basePath : String, parameterTypeService: ParameterT
+          def formatCondition(methodCall: MethodCall, parentBlock : List[MethodBlock]) =  {
+            (parentBlock.map(_.condition).filterNot(truthyCondition), truthyCondition(methodCall.condition)) match {
+              case (Nil, true) => "any"
+              case (list, true) => list.mkString("(",").(",")")
+              case (Nil, false) => methodCall.condition
+              case (list, false) => list.mkString("(", ").(", s".${methodCall.condition})")
             }
+          }
+          def canonifyCondition(methodCall: MethodCall) = {
+            formatCondition(methodCall, parentBlocks).replaceAll("""(\$\{[^}]*})""","""" + \$(Canonify-Class $1) + """")
+          }
 
-            def naReport(method : GenericMethod, expectedReportingValue : String) =
-              s"""_rudder_common_report_na ${componentName} -componentKey ${expectedReportingValue} -message "Not applicable" ${disableReporting} ${genericParams}"""
+          (for {
 
+            // First translate parameters to Dsc values
+            params    <- ((call.parameters.toList).traverse {
+                            case (id, arg) =>
+                              translater.translateToAgent(arg, AgentType.Dsc) match {
+                                case Full(dscValue) =>
+                                  parameterTypeService.translate(dscValue, methods.get(call.methodId).flatMap(_.parameters.find(_.id == id)).map(_.parameterType).getOrElse(ParameterType.StringParameter), AgentType.Dsc).map(dscValue => (id,dscValue))
+
+                                case eb : EmptyBox =>
+                                  val fail = eb ?~! s"Error when translating parameter '${arg}' of technique of method ${call.methodId} of technique ${techniqueName}"
+                                  Left(IOError(fail.messageChain,None))
+                              }
+                         }).map(_.toMap)
+
+            // Translate condition
+            condition <- translater.translateToAgent(canonifyCondition(call), AgentType.Dsc) match {
+                           case Full(c) => Right(c)
+                           case eb : EmptyBox =>
+                             val fail = eb ?~! s"Error when translating condition '${call.condition}' of technique of method ${call.methodId} of technique ${techniqueName}"
+                             Left(IOError(fail.messageChain,None))
+                         }
+
+            // Check if method exists
+            method <- methods.get(call.methodId) match {
+                        case Some(method) =>
+                          Right(method)
+                        case None =>
+                          Left(MethodNotFound(s"Method '${call.methodId.value}' not found when writing dsc Technique '${techniqueName}' methods calls", None))
+                      }
+            // Check if class parameter is correctly defined
+            classParameter <- params.get(method.classParameter) match {
```

Comment:
Can't you use this to also defined the classPrefix?

Reply:
it leads to `/bin/echo `"testing special characters ` è &é 'à é `"\`, so I don't think it's what you want